### ISSUE TITLE
useSuspenseQuery_experimental does not re-render the component despite the cache update

### ISSUE THREAD

**Issue by DamodarSojka** [apollographql/apollo-client#10653]
### Issue Description

client version: 3.8.0-alpha.8

1. This does refetch the query and updates the cache (checked via the dev tools). But does not re-render the component with the new data. Unmounting the component and re-mounting it (via route change, basically navigation to other screen without this component present) works correctly - new data is shown.

```
const { data } = useSuspenseQuery_experimental(getData, {
    variables: {
        someId,
    },
});

const [deleteFunction, { loading: loadingDelete, error: errorDelete }] = useMutation(deleteBuildingMutation, {
    refetchQueries: [{ query: getData, variables: { someId } }],
    awaitRefetchQueries: true, // both true and false result in the same behaviour
});
```

2. This works correctly, component re-renders with the new data.

```
const { data, refetch } = useSuspenseQuery_experimental(getData, {
    variables: {
        someId,
    },
});

const [deleteFunction, { loading: loadingDelete, error: errorDelete }] = useMutation(deleteBuildingMutation, {
    refetchQueries: [{ query: getData, variables: { someId } }],
    awaitRefetchQueries: true, // both true and false result in the same behaviour
    onCompleted: () => refetch(),
});
```

I understand this is experimental/alpha. If it is a known and expected behaviour the docs could mention that.

### Link to Reproduction

.

**Comment by jerelmiller** [comment]
Hey @DamodarSojka 👋 

This looks similar to https://github.com/apollographql/apollo-client/issues/10428 and https://github.com/apollographql/apollo-client/issues/10478. I've got 2 PRs to fix both issues (https://github.com/apollographql/apollo-client/pull/10601, https://github.com/apollographql/apollo-client/pull/10651). These will both be going out with the next alpha release.

Could you try the snapshot release from either PR to see if either fixes the issue?

* #10428 - `npm i @apollo/client@0.0.0-pr-10601-20230224210620`
* #10478 - `npm i @apollo/client@0.0.0-pr-10651-20230313210704`

**Comment by jerelmiller** [comment]
Hey @DamodarSojka 👋 

We released [`v3.8.0-alpha.10`](https://github.com/apollographql/apollo-client/blob/release-3.8/CHANGELOG.md#380-alpha10) today that contains these fixes. You can install the latest with `npm i @apollo/client@latest`. 

I'm going to go ahead and close this issue as a duplicate of #10478. If you find that the latest alpha doesn't address it, feel free to ping us again and we can dig in further.

Hope you're enjoying the alpha so far!
